lib/src/models/parsed_certificate.dart

```diff
@@ -74,19 +103,35 @@ sealed class ParsedCertificate with _$ParsedCertificate {
     required BigInt retirementEpoch,
   }) = StakePoolRetirement;
 
-  late final CertificateType certificateType = switch (this) {
-    StakeRegistration() => CertificateType.stakeRegistration,
-    StakeRegistrationConway() => CertificateType.stakeRegistrationConway,
-    StakeDeregistration() => CertificateType.stakeDeregistration,
-    StakeDeregistrationConway() => CertificateType.stakeDeregistrationConway,
-    StakeDelegation() => CertificateType.stakeDelegation,
-    VoteDelegation() => CertificateType.voteDelegation,
-    AuthorizeCommitteeHot() => CertificateType.authorizeCommitteeHot,
-    ResignCommitteeCold() => CertificateType.resignCommitteeCold,
-    DRepRegistration() => CertificateType.dRepRegistration,
-    DRepDeregistration() => CertificateType.dRepDeregistration,
-    DRepUpdate() => CertificateType.dRepUpdate,
-    StakePoolRegistration() => CertificateType.stakePoolRegistration,
-    StakePoolRetirement() => CertificateType.stakePoolRetirement,
+  late final bool isConway = switch (this) {
+    StakeRegistration() => false,
+    StakeDeregistration() => false,
+    StakeDelegation() => false,
+    StakePoolRegistration() => false,
+    StakePoolRetirement() => false,
+    StakeRegistrationConway() => true,
+    StakeDeregistrationConway() => true,
+    VoteDelegation() => true,
+    AuthorizeCommitteeHot() => true,
+    ResignCommitteeCold() => true,
+    DRepRegistration() => true,
+    DRepDeregistration() => true,
+    DRepUpdate() => true,
+  };
+
+  late final int certificateTypeSerializationValue = switch (this) {
+    StakeRegistration() => 0,
+    StakeDeregistration() => 1,
+    StakeDelegation() => 2,
+    StakePoolRegistration() => 3,
+    StakePoolRetirement() => 4,
+    StakeRegistrationConway() => 7,
+    StakeDeregistrationConway() => 8,
+    VoteDelegation() => 9,
+    AuthorizeCommitteeHot() => 14,
+    ResignCommitteeCold() => 15,
```

Comment:
There seem to be 4 more certificates (numbered 10 -> 13). Please double check again whether they exist or not in the typescript code.

P.s. There is a chance those other certificates do not appear in the TS sdk because they may not supported by ledger devices.

<img width="956" alt="image" src="https://github.com/vespr-wallet/ledger-cardano/assets/38853913/05c35ce8-8097-4a48-b016-3671f20a35ef">
Info from https://sancho.network/tools-resources/faq/

Reply:
all 13 of them are in the ts file : 

```export enum CertificateType {
  /**
   * Staking key registration certificate.
   * @see [[StakeRegistrationParams]]
   */
  STAKE_REGISTRATION = 0,
  /**
   * Staking key deregistration certificate
   * @see [[StakeDeregistrationParams]]
   */
  STAKE_DEREGISTRATION = 1,
  /**
   * Stake delegation certificate
   * @see [[StakeDelegationParams]]
   */
  STAKE_DELEGATION = 2,
  /**
   * Stake pool registration certificate
   * @see [[PoolRegistrationParams]]
   */
  STAKE_POOL_REGISTRATION = 3,
  /**
   * Stake pool retirement certificate
   * @see [[PoolRegistrationParams]]
   */
  STAKE_POOL_RETIREMENT = 4,
  /**
   * Staking key registration certificate with deposit
   * @see [[StakeRegistrationConwayParams]]
   */
  STAKE_REGISTRATION_CONWAY = 7,
  /**
   * Staking key deregistration certificate with deposit
   * @see [[StakeDeregistrationConwayParams]]
   */
  STAKE_DEREGISTRATION_CONWAY = 8,
  /**
   * Vote delegation certificate
   * @see [[VoteDelegationParams]]
   */
  VOTE_DELEGATION = 9,
  /**
   * Authorize constitutional committee member hot certificate
   * @see [[AuthorizeCommitteeParams]]
   */
  AUTHORIZE_COMMITTEE_HOT = 14,
  /**
   * Resign from constitutional committee certificate
   * @see [[ResignCommitteeParams]]
   */
  RESIGN_COMMITTEE_COLD = 15,
  /**
   * DRep registration certificate
   * @see [[DRepRegistrationParams]]
   */
  DREP_REGISTRATION = 16,
  /**
   * DRep deregistration certificate
   * @see [[DRepDeregistrationParams]]
   */
  DREP_DEREGISTRATION = 17,
  /**
   * DRep update certificate
   * @see [[DRepUpdateParams]]
   */
  DREP_UPDATE = 18,
}